public/js/services/game.js

```diff
@@ -1,6 +1,5 @@
 angular.module('mean.system')
-  .factory('game', ['socket', '$timeout', function (socket, $timeout) {
-
+  .factory('game', ['socket', '$timeout', '$http', function (socket, $timeout, $http) {
```

Comment:
You don't a new factory instance you can actually pass the $http service in the first factory instance